legal-api/src/legal_api/services/business_details_version.py

```diff
@@ -583,17 +589,32 @@ def party_role_revision_json(filing, party_role_revision, is_ia_or_after) -> dic
     @staticmethod
     def get_party_revision(filing, party_id) -> dict:
         """Consolidates all party changes up to the given transaction id."""
+        business = BusinessService.fetch_business_by_filing(filing)
+        if filing.legal_entity_id:
+            business_attr = LegalEntity
+            party_version = history_cls(business_attr)
+            business_attr_id = business_attr.id
+            party_version_business_id = party_version.id
+        else:
+            business_attr = AlternateName
+            party_version = history_cls(business_attr)
+            business_attr_id = business_attr.legal_entity_id
+            party_version_business_id = party_version.legal_entity_id
+            if business.is_owned_by_colin_entity:
+                business_attr_id = business_attr.colin_entity_id
+                party_version_business_id = party_version.colin_entity_id
+
         party_current = (
-            db.session.query(LegalEntity)
-            .filter(LegalEntity.change_filing_id == filing.id)
-            .filter(LegalEntity.id == party_id)
+            db.session.query(business_attr)
+            .filter(business_attr.change_filing_id == filing.id)
+            .filter(business_attr_id == int(party_id))
         )
 
-        party_version = history_cls(LegalEntity)
+        columns_to_select = VersionedBusinessDetailsService.select_revision_columns(party_version)
```

Comment:
I'm not sure if you are selecting the right columns for the party in all cases.  For example, it looks like if it's a SP person, SP DBA LEAR or SP DBA COLIN, you are selecting the alternate name columns.  I think you should be selecting the columns for the LE or AN record.